Xamarin.Forms.Core/MessagingCenter.cs

```diff
@@ -2,12 +2,30 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Reflection;
-using System.Runtime.CompilerServices;
 
 namespace Xamarin.Forms
 {
-	public static class MessagingCenter
+	public interface IMessagingCenter
 	{
+		void Send<TSender, TArgs>(TSender sender, string message, TArgs args) where TSender : class;
+
+		void Send<TSender>(TSender sender, string message) where TSender : class;
+
+		void Subscribe<TSender, TArgs>(object subscriber, string message, Action<TSender, TArgs> callback, TSender source = null) where TSender : class;
+
+		void Subscribe<TSender>(object subscriber, string message, Action<TSender> callback, TSender source = null) where TSender : class;
+
+		void Unsubscribe<TSender, TArgs>(object subscriber, string message) where TSender : class;
+
+		void Unsubscribe<TSender>(object subscriber, string message) where TSender : class;
+	}
+
+	public class MessagingCenter : IMessagingCenter
+	{
+		static readonly Lazy<MessagingCenter> s_instance = new Lazy<MessagingCenter>(() => new MessagingCenter());
+
+		public static IMessagingCenter Instance => s_instance.Value;
```

Comment:
Now that I think about it more, there really is no point. Initializing it in the static constructor is just as thread-safe and there's no startup performance gain because Core uses it for several features, so it always gets created when you start the application no matter what. 

I'll update it to use the static constructor.